dockyard/common/link.py

```diff
@@ -0,0 +1,20 @@
+
+def make_url(host='127.0.0.1', port=None, protocol='http', url=None):
+    head = protocol + '://'
+    head = head + host
+    if port: 
+        head = head + ":" + port
+
+    if url:
+        url = head + url
+    else:
+        url = head
+
+    return url
+
+def make_query_url(kwargs):
```

Comment:
May be something better can be done here as well. These methods does not seems to be flexible enough.